arrow-fx-coroutines/src/main/kotlin/arrow/fx/coroutines/Resource.kt

```diff
@@ -372,3 +393,179 @@ infix fun <A> Use<A>.release(release: suspend (A) -> Unit): Resource<A> =
  */
 infix fun <A> Use<A>.releaseCase(release: suspend (A, ExitCase) -> Unit): Resource<A> =
   Resource(acquire, release)
+
+/**
+ * Traverse this [Iterable] and collects the resulting `Resource<B>` of [f] into a `Resource<List<B>>`.
+ *
+ * ```kotlin:ank:playground
+ * import arrow.fx.coroutines.*
+ *
+ * class File(url: String) {
+ *   suspend fun open(): File = this
+ *   suspend fun close(): Unit {}
+ *   override fun toString(): String = "This file contains some interesting content!"
+ * }
+ *
+ * suspend fun openFile(uri: String): File = File(uri).open()
+ * suspend fun closeFile(file: File): Unit = file.close()
+ * suspend fun fileToString(file: File): String = file.toString()
+ *
+ * suspend fun main(): Unit {
+ *   //sampleStart
+ *   val res: List<String> = listOf(
+ *     "data.json",
+ *     "user.json",
+ *     "resource.json"
+ *   ).traverseResource { uri ->
+ *     resource {
+ *      openFile(uri)
+ *     } release { file ->
+ *       closeFile(file)
+ *     }
+ *   }.use { files ->
+ *     files.map { fileToString(it) }
+ *   }
+ *   //sampleEnd
+ *   res.forEach(::println)
+ * }
+ * ```
+ */
+inline fun <A, B> Iterable<A>.traverseResource(crossinline f: (A) -> Resource<B>): Resource<List<B>> =
+  fold(Resource.just(emptyList())) { acc: Resource<List<B>>, a: A ->
+    f(a).ap(acc.map { { b: B -> it + b } })
+  }
+
+/**
+ * Traverses and filters nullable resources
+ * @see traverseResource
+ */
+inline fun <A, B> Iterable<A>.traverseFilterResource(crossinline f: (A) -> Resource<B?>): Resource<List<B>> =
```

Comment:
Should we add a constraint on `B` to be `B: Any` to not allow users to make the mistake of returning null as a valid value only to have it being filtered out later? Don't think this is super relevant but I think this applies to some other places as well...